### DISCUSSION TITLE
GTK4 gem cannot be installed

### DISCUSSION THREAD

**Discussion by rexkogitans** [ruby-gnome/ruby-gnome#1452] [Q&A] [answered]
Sorry for asking a completely basic question, as I am new to Ruby (not new to programming in general).

I tried to install the `gtk3` gem first, and it was successful.

Then, I tried the same with gtk4 - in the meanwhile a gazillion of times -, and it failed, without any obvious reason. This is on Manjaro Linux.

````
$ gem install gtk4
Fetching gtk4-3.4.9.gem
Fetching gdk4-3.4.9.gem
Building native extensions. This could take a while...
ERROR:  Error installing gtk4:
	ERROR: Failed to build gem native extension.

    current directory: /home/rexkogitans/.local/share/gem/ruby/3.0.0/gems/gdk4-3.4.9/dependency-check
/usr/bin/ruby -I/usr/lib/ruby/3.0.0 -rrubygems /usr/lib/ruby/gems/3.0.0/gems/rake-13.0.6/exe/rake RUBYARCHDIR\=/home/rexkogitans/.local/share/gem/ruby/3.0.0/extensions/x86_64-linux/3.0.0/gdk4-3.4.9 RUBYLIBDIR\=/home/rexkogitans/.local/share/gem/ruby/3.0.0/extensions/x86_64-linux/3.0.0/gdk4-3.4.9

rake failed, exit code 1

Gem files will remain installed in /home/rexkogitans/.local/share/gem/ruby/3.0.0/gems/gdk4-3.4.9 for inspection.
Results logged to /home/rexkogitans/.local/share/gem/ruby/3.0.0/extensions/x86_64-linux/3.0.0/gdk4-3.4.9/gem_make.out
````

Can someone please help me?

**Comment by kou** [comment]
Could you attach `/home/rexkogitans/.local/share/gem/ruby/3.0.0/extensions/x86_64-linux/3.0.0/gdk4-3.4.9/gem_make.out`?

**Reply by rexkogitans** [reply]
It just contains this message once again, here is its content:
````
current directory: /home/rexkogitans/.local/share/gem/ruby/3.0.0/gems/gdk4-3.4.9/dependency-check
/usr/bin/ruby -I/usr/lib/ruby/3.0.0 -rrubygems /usr/lib/ruby/gems/3.0.0/gems/rake-13.0.6/exe/rake RUBYARCHDIR\=/home/rexkogitans/.local/share/gem/ruby/3.0.0/extensions/x86_64-linux/3.0.0/gdk4-3.4.9 RUBYLIBDIR\=/home/rexkogitans/.local/share/gem/ruby/3.0.0/extensions/x86_64-linux/3.0.0/gdk4-3.4.9

rake failed, exit code 1
````
This is my very problem: To me, this error message does not contain a useful message.

**Comment by kou** [comment, marked as answer]
What is your environment?

**Reply by rexkogitans** [reply]
What exactly do you mean by "environment"? Sorry for the silly question, since I am totally new to Ruby. Do you mean a certain environment variable, a config, the output of some command? Technically, it is a plain Manjaro Linux (Arch Linux derivative). I cannot install the gem neither globally (system-wide) nor per-user, both yields the same error. The example shows the per-user installation, while `gem install --no-user-install gtk4` just renders the base directory to be `/usr/lib/ruby/gems/3.0.0/gems/`.

**Reply by kou** [reply]
I wanted to know how to reproduce this on my local environment. I'm on Debian GNU/Linux sid.
Is there a Docker image for Manjaro Linux to reproduce this?

**Reply by rexkogitans** [reply]
https://github.com/manjaro/manjaro-docker seems to be a vivid Docker project providing Manjaro Linux. I have not tested it. However, you can always run [plain Manjaro Linux](https://wiki.manjaro.org/index.php/Download_Manjaro) in a fully virtualised environment, like VirtualBox.

**Reply by kou** [reply]
Thanks. https://hub.docker.com/r/manjarolinux/base works.

It seems that GTK4 doesn't provide `gdk-4.0` pkg-config package. We should integrate gdk4 gem into gtk4 gem. Please wait for the next release.

**Reply by rexkogitans** [reply]
@kou When, approximately, can a new relase be expected?

**Reply by kou** [reply]
I hope that we can release a new version in a few weeks but we can't include gtk4 gem related improvement in the next release. Sorry.
We work on it after the next release.
